app/Database/migrations/2023_12_15_154815_create_user_oauth_tokens_table.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
+
+return new class() extends Migration {
+    /**
+     * Run the migrations.
+     */
+    public function up(): void
+    {
+        Schema::create('user_oauth_tokens', function (Blueprint $table) {
+            $table->id();
+            $table->unsignedInteger('user_id');
+            $table->string('provider');
+            $table->string('token');
+            $table->string('refresh_token');
+            $table->dateTime('last_refreshed_at')->nullable();
+            $table->timestamps();
+
+            $table->foreign('user_id')
```

Comment:
I am not sure about using foreign keys in migrations and tie tables to each other on database level.

Don't we have model relationships between User and Oauth models ?